tensorflow/python/kernel_tests/decode_image_op_test.py

```diff
@@ -25,84 +25,62 @@
 class DecodeImageOpTest(tf.test.TestCase):
 
   def testGif(self):
-    # GIF encoded 2x2 image
-    image_bytes = (b'GIF89a\x02\x00\x02\x00\xf1\x01\x00\x00\x00\x00\xff\xff\xff'
-                   b'\xff\xff\xff\x00\x00\x00!\xf9\x04\x00\x00\x00\x00\x00,\x00'
-                   b'\x00\x00\x00\x02\x00\x02\x00\x00\x02\x03D4\x05\x00;')
-    expected_values = np.array([[[[  0,   0,   0], [255, 255, 255]],
-                                [[255, 255, 255], [0, 0, 0]]]], 
-                                dtype=np.uint8)
-    decode = tf.image.decode_image(image_bytes)
-    with self.test_session():
-      decoded_pixels = decode.eval()
-      self.assertAllEqual(decoded_pixels, expected_values)
+    # Read some real GIFs
+    prefix = 'tensorflow/core/lib/gif/testdata/'
+    filename = 'scan.gif'
+    WIDTH = 20
+    HEIGHT = 40
+    STRIDE = 5
+    shape = (12, HEIGHT, WIDTH, 3)
+
+    with self.test_session(use_gpu=True) as sess:
+      gif0 = tf.read_file(prefix + filename)
+      image0 = tf.image.decode_image(gif0)
+      image1 = tf.image.decode_gif(gif0)
+      gif0, image0, image1 = sess.run([gif0, image0, image1])
+
+      self.assertEqual(image0.shape, shape)
+      self.assertAllEqual(image0, image1)
+
+      for frame_idx, frame in enumerate(image0):
+        gt = np.zeros(shape[1:], dtype=np.uint8)
+        start = frame_idx * STRIDE
+        end = (frame_idx + 1) * STRIDE
+        if end <= WIDTH:
+          gt[:, start:end, :] = 255
+        else:
+          start -= WIDTH
+          end -= WIDTH
+          gt[start:end, :, :] = 255
```

Comment:
I grabbed this from the GIF tests in python/ops/image_ops_test.py. The [test image](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/lib/gif/testdata/scan.gif) is a multi-framed image of a white bar travelling across a black background. It looks like this test starts with an all-black (zeros) array and then fills in the appropriate pixels with white (255) depending on which frame it is.

I can comment an explanation in or remove this check if you want.